src/ai/backend/manager/models/vfolder.py

```diff
@@ -108,6 +110,29 @@ class VFolderInvitationState(str, enum.Enum):
     REJECTED = "rejected"  # rejected by invitee
 
 
+class VFolderOperationStatus(str, enum.Enum):
+    """
+    Introduce virtual folder current status for storage-proxy operations.
+    """
+
+    READY = "ready"
+    PERFORMING = "performing"
+    CLONING = "cloning"
+    DELETING = "deleting"
+    DELETED = "deleted"
+    MOUNTED = "mounted"
+
+
+class VFolderAccessStatus(str, enum.Enum):
+    """
+    Introduce virtual folder desired status for storage-proxy operations.
+    Not added to db scheme  and determined only by current vfolder status.
+    """
+
+    READABLE = "readable"
+    WRITABLE = "writable"
```

Comment:
Hmm, how about changing name of this enum to `UPDATABLE` instead of `WRITABLE`? There can be a single chance that others can misunderstand this status as a 'status that user can write file to this vFolder'.